doc/dune-files.rst

```diff
@@ -69,6 +69,8 @@ dialect
    .. dune:field:: implementation
 
       Details related to the implementation files (corresponding to `*.ml`).
+      
+      In Dune 3.9 and later, this field is optional.
```

Comment:
Probably should mention dune lang rather than dune since newer versions of dune will have the old behaviour if the lang version is not up to date.

Reply:
done